server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -1197,6 +1197,10 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
             }
 
             if (gateway != null && netmask != null) {
+                if(!NetUtils.isNetworkorBroadcastIP(gateway,netmask)) {
+                    s_logger.debug("The gateway IP provided is "+ gateway + " and netmask is "+ netmask + ". The IP is either broadcast or network IP.");
```

Comment:
maybe you can change the name of the static object to LOGGER while you are at it?